modules/shapefile/src/lib/parsers/parse-shapefile.js

```diff
@@ -0,0 +1,183 @@
+import {binaryToGeoJson} from '@loaders.gl/gis';
+import {parseShx} from './parse-shx';
+import {zipBatchIterators} from '../streaming/zip-batch-iterators';
+import {SHPLoader} from '../../shp-loader';
+import {DBFLoader} from '../../dbf-loader';
+
+export async function* parseShapefileInBatches(asyncIterator, options, context) {
+  const {parseInBatches, fetch, url} = context;
+  const {shx, cpg, prj} = await loadShapefileSidecarFiles(options, context);
+
+  // parse geometries
+  const shapeIterator = await parseInBatches(asyncIterator, SHPLoader); // {shp: shx}
+  const shapeHeader = await shapeIterator.next();
+
+  // parse properties
+  let propertyIterator;
+  let generator;
+  try {
+    const dbfResponse = await fetch(replaceExtension(url, 'dbf'));
+    if (dbfResponse.ok) {
+      propertyIterator = await parseInBatches(dbfResponse, DBFLoader, {
+        dbf: {encoding: cpg || 'latin1'}
+      });
+      generator = await zipBatchIterators(shapeIterator, propertyIterator);
+    } else {
+      // Ignore properties
+      generator = shapeIterator;
+    }
+  } catch (error) {
+    // Ignore properties
+    generator = shapeIterator;
+  }
```

Comment:
I'm sure there's a better way to fall back than the double-nested `generator = shapeIterator`, and the `if (!propertyIterator)` block below looks a little bad.